### ISSUE TITLE
Use both key_prefix and query_string to form the cache key

### ISSUE THREAD

**Issue by Ashwin-shankar50** [pallets-eco/flask-caching#302]
### Description
In current implementation **_cached_** function under the **_Cache_** class considers either _**query_string**_ parameter or **_key_prefix_** parameter to build the cache key 

It can be enhanced and made to consider both options together to build the cache key. 

### Explanation

If the cache key needs to be built like,

**Expectation** -
_Expected format_ - `flask_cache_<key_prefix>/<request_url><query_string_hash>`
_Example_ - `flask_cache_user_managemen//api/v1/users52440c19267b11c4bbe10e97c6e45885`

**Current possibilities** -

_Format 1_ - `flask_cache_<key_prefix>/<request_url>`
_Example_ - `flask_cache_user_management//api/v1/users `

_Format 1_ - `flask_cache_<request_url><query_string_hash>`
_Example_ - `flask_cache_/api/v1/users52440c19267b11c4bbe10e97c6e45885`

**Comment by snarfed** [comment]
Agreed! Not just a feature request but arguably a bug fix, or at least prevention. There's currently no indication that `query_string` silently overrides `key_prefix` and ignores it, which is an unpleasant surprise.

(Thank you for maintaining flask-caching @northernSage et al, it's great!)

**Comment by nightcityblade** [comment]
Hi, I'd like to work on this. I'll submit a PR shortly.
